doc/tests/README.md

```diff
@@ -1,63 +1,73 @@
+
 The testing approach for PermaplanT will follow a mix of unit testing, integration testing, and manual testing.
 Unit testing will be conducted for individual components and functions to ensure that they are working as expected.
 Integration testing will be conducted for API endpoints and interactions between components to ensure that they are functioning correctly together.
 Manual testing will be conducted for end-to-end functionality and overall user experience.
 
-## Unit Testing
+## Objectives
 
-Unit tests are used to test individual units of code in isolation from the rest of the system. This is important because it allows us to validate that each unit is working as intended, without interference from other parts of the system.
+- Ensure that the software meets functional requirements and operates as intended.
+- Identify and mitigate risks associated with software defects and failures.
+- Enhance user experience by maintaining low bugrate and responsiveness.
+- Improve overall software quality and reliability.
+- Maintain an 80% backend branch coverage.
 
-### Frontend
+## Out of Scope:
 
-For the frontend unit tests will be used to test the following areas:
+- Database tests
+- Stress tests
 
-- State management logic: Since the frontend relies heavily on state management, it is important to ensure that the state is being properly managed and updated.
-- Functions that handle data manipulation, such as sorting or filtering
-- Components that render UI elements
-- Async actions, such as API calls or event handling
-- Form validation and submission
+## In Scope:
 
-### Backend
+- End-to-End testing of critical workflows
+- Integration testing of various modules and components
+- System tests
+- Performance tests
+- Security tests
+- Regression testing
+- User acceptance testing
+- Compatibility testing on different platforms, browsers, and devices?
 
-For the backend, unit tests will be used to test the following areas:
+## Testing Types and Levels
 
-- Database queries and operations
-- Business logic and data manipulation, e.g. code that performs calculations, manipulates data, or makes decisions based on input
-- Authentication and authorization
-- Error handling
+### 1. Unit Testing
 
-In the backend unit tests can be found in the `src/` directory.
-They are to be put in the same file as the functions or modules they are supposed to test.
-Create separate modules called test in that file annotated with `#[cfg(test)]` (that way they are only compiled when actually running tests).  
-See the [rust doc](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html) for reference.
+Unit tests are used to test individual units of code in isolation from the rest of the system. This is important because it allows us to validate that each unit is working as intended, without interference from other parts of the system.
```

Comment:
```suggestion
Unit tests are used to test individual units of code in isolation from the rest of the system. 
This is important because it allows us to validate that each unit is working as intended, without interference from other parts of the system.
```
Our documentation guidelines state that there should be one sentence per line.

Reply:
I missed that one, got it now, thank you.